common/beerocks/tlvf/AutoGenerated/include/beerocks/tlvf/beerocks_message_1905_vs.h

```diff
@@ -30,7 +30,7 @@ class tlvVsClientAssociationEvent : public BaseClass
 {
     public:
         tlvVsClientAssociationEvent(uint8_t* buff, size_t buff_len, bool parse = false);
-        tlvVsClientAssociationEvent(std::shared_ptr<BaseClass> base, bool parse = false);
+        explicit tlvVsClientAssociationEvent(std::shared_ptr<BaseClass> base, bool parse = false);
```

Comment:
@tomereli Do you know why the autogenerated code is even in git?
Usually, autogenerated code should not be put under version control. There's no point and it can cause conflicts in some cases (in our case, it just clutters up reviews that change the TLVs)

Reply:
If everyone new to the project asks this question, maybe there's something wrong with the answer :-)

The answer is:

-----

By putting the AutoGenerated source in revision control, it is easier for someone to browse the code after a simple checkout or even online, without building it. Even for review it is sometimes easier, e.g. when tlvf is changed it's more relevant to review the generated code than the python code.

And to make sure the AutoGenerated code in git is always consistent with the yaml source, we have CI that verifies it.

-----

I also wouldn't have chosen to put the generated code in git, but I don't feel so strongly about it that I'm going to fight the decision. I believe @vitalybu is the only real proponent.

Reply:
What Arnout said is the answer and I fully agree with it. If you want we can put it up for a vote again..

Reply:
I actually was against it at first as well, but the reasons Arnout mentioned eventually convinced me.

Reply:
Hang on, so who actually wants the generated code in git? IIUC @tomereli, @vitalybu, @arnout and @fvbogaert are all saying that they would prefer the generated code to *not* be in git, but that they see the arguments and are not opposed to it either. So who is left as a proponent? @morantr ? @rmelotte ?

Just to be clear, some advantages of *not* having the generated code in git:

- There is no risk that the generated code is inconsistent with the yaml files.
- It makes it clearer that this is generated code that shouldn't be modified.
- It removes some commits ("update autogenerated") and makes others easier to review.
- We can remove the check from CI.

Reply:
I don't like having the generated code in the repository either.
However in this case, the way to generate it is nothing "standard" or easy to read or check for errors, so I also agree for reviews it can help to have it included.

Reply:
@arnout - The main disadvantage of removing the autogenerated files from git is reviewing changes to tlvf.py. Unfortunately, tlvf.py is not a very readable python module, so I think that a followup commit which shows the change is an advantage, not a disadvantage.
I assume @morantr and @LiorAmram would agree on this.

Reply:
M2c for this discussion -
Some PDFs which are the output of a complicated 'compilation' process are kept in source control. E.g. https://github.com/wg21il/Papers/tree/P2022/master/P2022/generated
These files are the "final product" and are the subject for review by humans.

Intermediate code, of all kinds, that is meant to be consumed mainly by other code, IMO should be kept locally, not in a source control, the reasons @arnout popinted.

The main issue that I see, which legitimate the use of git for generated code is human review.
So maybe we are using an incorrect tool to solve the problem.
Maybe we should address the review-auto-generated-code problem differently.
E.g. Supply a really easy-to-use generator.

Reply:
OK @RanRegev, which easy-to-use generator do you have in mind?

Reply:
@tomereli, 
1. I suggested that sometimes we choose the wrong tool to solve a real problem.
If we agree that the main reason to save auto generated files in git is to enable humans review them - then maybe it is the wrong tool.
2. As for your question:
a. the [documentation](https://github.com/prplfoundation/prplMesh/blob/master/framework/tlvf/README.md) does not help. It says that there are 2 argument needed, but it calls for at least 3. I spent 10 minutes on it and I still don't know how to use tlvf.py and compile even a single file. Yes, maybe I'm a slow catcher and hard to understand, but 10 minutes are too much.
b. I understand the that _final_ requirement is that this script runs via the build system: bring other headers with it, compile many files together, have debug flags, and more. But this use of the script doesn't mean that it has to be complicated from the first place. For simple review it will be enough to have `tlvf.py <src-yaml>` - and that it is. No configuration file, not even an output file (stdout is great).
On top of this simplicity we can add as much complexity as we want. 
Having broad ranges of opportunities to compile tlv may, maybe, will remove the need store products in git.

Reply:
> @tomereli,
> 
> 1. I suggested that sometimes we choose the wrong tool to solve a real problem.
>    If we agree that the main reason to save auto generated files in git is to enable humans review them - then maybe it is the wrong tool.

I still don't understand your point - what is the problem you are referring to and what is the correct tool in your opinion we should use?

> 2. As for your question:
>    a. the [documentation](https://github.com/prplfoundation/prplMesh/blob/master/framework/tlvf/README.md) does not help. It says that there are 2 argument needed, but it calls for at least 3. I spent 10 minutes on it and I still don't know how to use tlvf.py and compile even a single file. Yes, maybe I'm a slow catcher and hard to understand, but 10 minutes are too much.

The tlvf generates (not compiles) multiple files, there is no support to generate a single file. We didn't hear any complaints till now about TLVF from other developers. Perhaps if you struggle with TLVF than ask someone from your team to guide you, almost everyone in the community knows how to add a new class / structure to TLVF. Raising complaints or arguments that TLVF is hard to use is pointless - we have plans for TLVFv2 but its a huge effort and since TLVFv1 just works, there is no plan to replace it anytime soon. Its not like we had a wide array of tools to select from and we chose TLVF - there was none.

>    b. I understand the that _final_ requirement is that this script runs via the build system: bring other headers with it, compile many files together, have debug flags, and more. But this use of the script doesn't mean that it has to be complicated from the first place. For simple review it will be enough to have `tlvf.py <src-yaml>` - and that it is. No configuration file, not even an output file (stdout is great).
>    On top of this simplicity we can add as much complexity as we want.
>    Having broad ranges of opportunities to compile tlv may, maybe, will remove the need store products in git.

Even if this was supported by TLVF, it wouldn't help to review TLVF changes. IMO nothing we change in TLVF will make any difference - unless we trust TLVF as a complete black box, then we can just use the YAML files and ignore the autogenerated files altogether. We are actually pretty close to this now, TLVF supports almost everything we need.

Regarding your suggestion for having tlvf option which accepts a single yaml file - this is not possible since yaml files may refer to other yaml files, and in most cases they do. Even it would work (and it wouldn't) - it won't make reviewing code easier, since it will cause developers to run tlvf to generate code instead of just browsing it, which beats the purpose IMO.

Reply:
Fine.
Again - I'm not complaining on tlvf.
The question in this discussion is *why do we keep products in git?*
I see lots of drawbacks doing so, and I am not sure git is the correct solution to the problem raised when it was chosen. _I might be wrong_ by not understanding the problem from the first place.
Currently I don't work with tlvs, change them or have anything with them. When I will the _last_ thing that I'll do is to suggest a change in the mechanism. And that I will do *only after* I have tried everything with all the knowledge I can get.
Suggesting changing without understanding the problem is a _huge_ mistake. I'm not doing that.
The point here is not the tlv - but saving compilation (a.k.a generated) results.